core/src/syscall/precompiles/keccak256/air.rs

```diff
@@ -64,34 +65,16 @@ where
         );
 
         // Constrain that the inputs stay the same throughout the 24 rows of each cycle
-        let mut not_last_in_cycle = local.keccak.step_flags[0].into();
-        for i in 1..NUM_ROUNDS - 1 {
-            not_last_in_cycle = not_last_in_cycle + local.keccak.step_flags[i];
-        }
-        builder
-            .when_transition()
-            .when(not_last_in_cycle.clone() * local.is_real)
-            .assert_eq(local.shard, next.shard);
-        builder
-            .when_transition()
-            .when(not_last_in_cycle.clone() * local.is_real)
-            .assert_eq(local.clk, next.clk);
-        builder
-            .when_transition()
-            .when(not_last_in_cycle.clone() * local.is_real)
-            .assert_eq(local.state_addr, next.state_addr);
-        builder
-            .when_transition()
-            .when(not_last_in_cycle)
-            .assert_eq(local.is_real, next.is_real);
```

Comment:
Just removed filtering by `is_real`, I think dummy rows will still satisfy these consistency checks since they seem to consistently have zeros for these fields?

Reply:
Indeed! they do have zeros